crates/iceberg/src/avro/schema.rs

```diff
@@ -826,7 +914,117 @@ mod tests {
                 .unwrap()
         };
 
-        check_schema_conversion(avro_schema, iceberg_schema, false);
+        check_schema_conversion(avro_schema, iceberg_schema);
+    }
+
+    #[test]
+    fn test_schema_with_array_map() {
+        let avro_schema = {
+            AvroSchema::parse_str(
+                r#"
+{
+    "type": "record",
+    "name": "avro_schema",
+    "fields": [
+        {
+            "name": "optional",
+            "type": {
```

Comment:
Can you add the `element-id` here as well?

Reply:
Or maybe better, to add two new tests:

- One being an Avro map (not the `List` of `Structs` that's Iceberg specific to have maps with keys other than strings)
- One being an Avro list

Reply:
> Can you add the element-id here as well?

I'm a little confused with this element id. Where is this element-id when we convert this schema back to iceberg. We will convert this schema back to a map in the schema so there is no place to use element-id. Or is it just a placeholder?🤔

Reply:
Ah, it is actually here: https://github.com/apache/iceberg-rust/pull/411/files#diff-542e38433ab477bbc8669524448ecf923f3f0b12c52d725984d1d160e536656eR958
Sorry for the confusion 👍